### ISSUE TITLE
Webhook for filled orders

### ISSUE THREAD

**Issue by nickjuntilla** [rarible/protocol#3]
We would like to alert our users when an NFT is sold. This way we could send them a congratulatory email or something of that nature. This could be possible by having a webhook callback parameter in the sell order.

**Comment by antho1404** [comment]
Currently, one workaround is to [listen to to the event](https://docs.ethers.io/v5/api/contract/contract/#Contract--events) [`Match`](https://github.com/rarible/protocol-contracts/blob/b7777cc2566129b63d783652fe55efcd91b6862a/exchange-v2/contracts/ExchangeV2Core.sol#L25) of the exchange smart contract end send an email or whatever job you need when the event is received.

That's not ideal as this requires your backend to maintain a connection to the blockchain but at least there is a way to do that.

**Comment by evgenynacu** [comment]
One question: do you prefer to get a webhook for this or for example to use websocket for event streaming?

**Comment by antho1404** [comment]
Webhook would definitely be better as it can be connected to a stateless/serverless endpoint.
Websocket is interesting but maintaining an open connection is always tricky. I don't see much value in that, as it can already be done by directly listening to the events in the smart contract.

**Comment by nickjuntilla** [comment]
I agree with @antho1404 we use stateless google cloud functions and the webhook is great.

**Comment by evgenynacu** [comment]
Ok, thanks for the information @antho1404 @nickjuntilla 

Also, if you have capacity to contribute, will guide you to how this can be implemented and what should be changed.
Let me know if you are ready to contribute

**Comment by antho1404** [comment]
Depending on the time required, I might be able to contribute to that.

**Comment by evgenynacu** [comment]
@sanufriev could you guide @antho1404 and anyone interested what's needed to be done to add a webhook for order state change

**Comment by antho1404** [comment]
This issue has been closed. What's the status? Is this something possible to do now? If so, how? Or is this something that will not be done because it can be done directly by listening to the chain?

**Comment by eduardstal** [comment]
> This issue has been closed. What's the status? Is this something possible to do now? If so, how? Or is this something that will not be done because it can be done directly by listening to the chain?

Hey Antho, I was under the mistaken idea that you were reached by @sanufriev already. Then this issue should not be closed, let me try to get ahold of him again to look over this issue.
